pkg/pipeline/decode/decode_aws.go

```diff
@@ -51,11 +53,12 @@ func (c *decodeAws) Decode(in []interface{}) []config.GenericMap {
 	out := make([]config.GenericMap, 0)
 	nItems := len(in)
 	log.Debugf("nItems = %d", nItems)
-	for _, line := range in {
+	for i, line := range in {
 		lineSlice := strings.Fields(line.(string))
 		nFields := len(lineSlice)
-		if nFields > len(c.keyTags) {
-			nFields = len(c.keyTags)
+		if nFields != len(c.keyTags) {
+			log.Errorf("decodeAws Decode: wrong number of fields in line %d", i+1)
+			continue
 		}
 		record := make(config.GenericMap)
 		for i := 0; i < nFields; i++ {
```

Comment:
can you use significant name and not `i`

Reply:
OK